contributors/design-proposals/storage/pv-monitoring-proposal.md

```diff
@@ -0,0 +1,128 @@
+# PV monitoring proposal
+
+Status: Pending
+
+Version: Alpha
+
+Implementation Owner: NickrenREN@ 
+
+## Motivation
+
+For now, kubernetes has no way to monitor the PVs, which may cause serious problems. 
+For example: if volumes are unhealthy, and pods do not know that and still try to get and write data, 
+which will lead to data loss and unavailability of services. 
+So it is necessary to have a mechanism for monitoring PVs and react when PVs have problems.
+
+## Proposal
+
+We can separate the proposal into two parts:
+
+* monitoring PVs and marking them if they have problems
+* reacting to the unhealthy PVs
+
+For monitoring, we may create a controller for it, and each volume plugin should have its own function to check volume health. 
+Controller can call them periodically. The controller also needs to watch node events because local PVs will be unreachable if nodes break down.
+
+For reacting, different kinds of apps may have different methods,we can also create a controller for it. 
+
+At first phase, we may just react to statefulsets with local storage PVs.
+
+## User Experience
+### Use Cases
+
+* Users create many local PVs manually and want to monitor their health condition;
```

Comment:
Since all of the use cases here are for local PVs, should we just have this document be specific to local PVs for now?  We can consider extending this to other PV types in the future if the need arises.

Reply:
yeah, we want to focus on local storage at the first stage, and then move to other storage drivers if needed.